### PR TITLE
Adding function to allow for updating client version

### PR DESCRIPTION
**Pull request by nishanttotla**
This is related to the discussion in #174 

Signed-off-by: Nishant Totla <nishanttotla@gmail.com>

### CODE REVIEW

**Comment by dongluochen** [comment]
LGTM

**Comment by duglin** [comment]
I know its a trivial change/function - but can we get a testcase for this?

**Comment by nishanttotla** [comment]
@duglin did you have something particular in mind? There's no logic in the function, so I'm not sure what we'd test.

**Comment by duglin** [comment]
I was thinking of making sure that the version string appears correctly in the outgoing http request.

**Comment by nishanttotla** [comment]
@duglin Just added tests. Please take a look :)

File: client/client_test.go [hunk 1]
```diff
@@ -70,3 +70,30 @@ func TestParseHost(t *testing.T) {
...
+	}
+
+	apiClient, err := NewClient("unix:///var/run/docker.sock", "", nil, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, cs := range cases {
+		apiClient.UpdateClientVersion(cs.v)
+		g := apiClient.getAPIPath(cs.p, cs.q)
```
Comment (duglin) [inline client/client_test.go]: My concern with checking getAPIPath instead of actually checking what the server sees is that if someone adds a new code path to open a connection then we'll miss it.  Might not be a big deal but I'd like to hear what @vdemeester thinks.
See https://github.com/docker/engine-api/blob/master/client/container_create_test.go for an example of checking it from the server side.

File: client/client_test.go [hunk 2]
```diff
@@ -70,3 +70,30 @@ func TestParseHost(t *testing.T) {
...
+
+func TestUpdateClientVersion(t *testing.T) {
+	cases := []struct {
+		v string
+		p string
+		q url.Values
+		e string
+	}{
+		{"1.20", "/containers/json", nil, "/v1.20/containers/json"},
+		{"v1.21", "/containers/json", url.Values{}, "/v1.21/containers/json"},
```
Comment (duglin) [inline client/client_test.go]: Interesting that w/ and w/o a leading "v" will generate the same result - not sure I would have done that, but that's not part of this PR  :-)

Comment (nishanttotla) [inline client/client_test.go]: Yeah, I just looked at the test above, and assumed it was required/expected behavior.

File: client/client_test.go (hunk 1 above)
Comment (vdemeester) [inline client/client_test.go]: I do agree with @duglin, I think it would better to check *server side* :angel:

**Comment by nishanttotla** [comment]
@duglin @vdemeester updated, please let me know if this works!

File: client/client_test.go [hunk 3]
```diff
@@ -70,3 +78,43 @@ func TestParseHost(t *testing.T) {
 		}
 	}
 }
+
+func TestUpdateClientVersion(t *testing.T) {
+	client := &Client{
+		transport: newMockClient(nil, func(req *http.Request) (*http.Response, error) {
+			reqString := req.URL.String()
+			splitQuery := strings.Split(reqString, "/")
+			queryVersion := splitQuery[3]
```
Comment (duglin) [inline client/client_test.go]: I think you'd be better of using req.URL.Path instead of req.URL.String() then you don't need to worry about any leading /'s that are before the `/vX.Y` part of the URL.  And then you know that the version part is always the first thing in that string.

**Comment by duglin** [comment]
One minor request on the server-side parsing of the incoming URL.
Aside from that LGTM

**Comment by vdemeester** [comment]
Same nit as @duglin, otherwise LGTM 🐰

**Comment by nishanttotla** [comment]
@duglin @vdemeester that makes sense, thanks. It's updated now.

**Comment by duglin** [comment]
still LGTM
ping @vdemeester for the re-LGTM and merge

**Comment by vdemeester** [comment]
ah yes, re-LGTM :angel:
